File: kotlin-core-maven/src/test/java/org/example/kotlin/core/java/ConstantsTest.java
package org.example.kotlin.core.java;

import org.example.kotlin.core.java.model.SomeAnnotation;
import org.example.kotlin.core.kotlin.model.Constants;
import org.example.kotlin.core.kotlin.model.ConstantsKt;
import org.junit.Test;

public class ConstantsTest {

    @SomeAnnotation(text = ConstantsKt.CONST_VAL)
    public static class SomeAnnotatedClass {
    }
    @SomeAnnotation(text = Constants.CONST_VAL)
    public static class SomeAnnotatedClass2 {
    }

    @Test
    public void compileTimeConstants() {
        System.out.printf("annotation: %s%n", SomeAnnotatedClass.class.getAnnotation(SomeAnnotation.class).text());
        System.out.printf("annotation2: %s%n", SomeAnnotatedClass2.class.getAnnotation(SomeAnnotation.class).text());
    }

}

File: kotlin-core-maven/src/test/kotlin/org/example/kotlin/core/kotlin/ConstantsTest.kt
package org.example.kotlin.core.kotlin

import org.example.kotlin.core.java.model.SomeAnnotation
import org.example.kotlin.core.kotlin.model.CONST_VAL
import org.example.kotlin.core.kotlin.model.Constants
import org.junit.Test

class ConstantsTest {
    @SomeAnnotation(text = CONST_VAL)
    class SomeAnnotatedClass {
    }
    @SomeAnnotation(text = Constants.CONST_VAL)
    class SomeAnnotatedClass2 {
    }

    @Test
    fun compileTimeConstants() {
        println("annotation: ${SomeAnnotatedClass::class.java.getAnnotation(SomeAnnotation::class.java).text}")
        println("annotation2: ${SomeAnnotatedClass2::class.java.getAnnotation(SomeAnnotation::class.java).text}")
    }
}